sample-app/src/main/java/com/joinforage/android/example/ui/pos/ReceiptView.kt

```diff
@@ -0,0 +1,163 @@
+    orientation = LinearLayout.VERTICAL
+
+    layoutParams = LinearLayout.LayoutParams(
+        LinearLayout.LayoutParams.MATCH_PARENT,
+        LinearLayout.LayoutParams.WRAP_CONTENT
+    )
+
+    // Set background color
+    val backgroundColor = ContextCompat.getColor(context, R.color.light_grey)
+    setBackgroundColor(backgroundColor)
+
+    // Convert 8dp padding to pixels
+    val paddingInPixels = (8 * context.resources.displayMetrics.density).toInt()
+    setPadding(paddingInPixels, paddingInPixels, paddingInPixels, paddingInPixels)
+
+    // add the TextViews that make up the ReceiptDisplay's content
+    receiptLayout.lines.forEach { line ->
+        val lineLayout = createReceiptLineLinearLayout(context, line)
+        addView(lineLayout)
+    }
+}
+
+class ReceiptView(
+    context: Context
+) : ScrollView(context) {
+    private val scrollableContent: LinearLayout
+    private val printReceiptBtn: Button
+    private var receiptDisplay: LinearLayout
+    private var receiptLayout = ReceiptLayout.EmptyReceiptLayout
+
+    init {
+        // organize the scrollview itself will house (the root)
+        // which will house the container that will overflow as scroll
+        layoutParams = LinearLayout.LayoutParams(
+            ConstraintLayout.LayoutParams.MATCH_PARENT,
+            ConstraintLayout.LayoutParams.WRAP_CONTENT
+        )
+        val rootPaddingInPixels = (16 * context.resources.displayMetrics.density).toInt()
+        setPadding(rootPaddingInPixels, rootPaddingInPixels, rootPaddingInPixels, rootPaddingInPixels)
+
+        // organize and add the subviews
+        scrollableContent = LinearLayout(context).apply {
+            orientation = LinearLayout.VERTICAL
+            layoutParams = LinearLayout.LayoutParams(
+                LinearLayout.LayoutParams.MATCH_PARENT,
+                LinearLayout.LayoutParams.WRAP_CONTENT
+            )
+
+            printReceiptBtn = Button(context).apply {
+                layoutParams = LinearLayout.LayoutParams(
+                    LinearLayout.LayoutParams.WRAP_CONTENT,
+                    LinearLayout.LayoutParams.WRAP_CONTENT
+                )
+                text = "Print Receipt"
+
+                // when the button is clicked, have it pass the current
+                // receiptLayout (datastructure) to our ReceiptPrinter
+                // service so it so we can print it on the POS terminal
+                setOnClickListener {
+                    val cpayPrinter = DevicesFactory.getDeviceManager().printDevice
+                    ReceiptPrinter(receiptLayout).printWithCPayTerminal(cpayPrinter)
+                }
+            }
+            addView(printReceiptBtn)
+
+            // organize the linear layout that will hold the receipt details
+            receiptDisplay = createReceiptDisplay(context, receiptLayout)
+            addView(receiptDisplay)
+        }
+        addView(scrollableContent)
+        println("constructor was called!!")
```

Comment:
can we remove this `println`

Reply:
this one can be nuked for sure! Good catch!

Reply:
https://github.com/teamforage/forage-android-sdk/pull/148/commits/39149b28e4d57c93fab9d3ea3c7c17a7e5b2b098